86BoxManager/Settings/Settings.cs

```diff
@@ -0,0 +1,276 @@
+            }
+            if (!File.Exists(EXEDir + "86Box.exe") && !File.Exists(EXEDir + @"\86Box.exe"))
+            {
+                DialogResult result = MessageBox.Show("86Box.exe could not be found in the directory you specified, so you won't be able to use any virtual machines. Are you sure you want to use this path?", "Warning", MessageBoxButtons.YesNo, MessageBoxIcon.Warning);
+                
+                if (result == DialogResult.No)
+                {
+                    return false;
+                }
+            }
+            try
+            {
+                RegistryKey regkey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\86Box", true); //Try to open the key first (in read-write mode) to see if it already exists
+                
+                if (regkey == null) //Regkey doesn't exist yet, must be created first and then reopened
+                {
+                    Registry.CurrentUser.CreateSubKey(@"SOFTWARE\86Box");
+                    regkey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\86Box", true);
+                    regkey.CreateSubKey("Virtual Machines");
+                }
+
+                //Store the new values, close the key, changes are saved
+                regkey.SetValue("EXEdir", EXEDir, RegistryValueKind.String);
+                regkey.SetValue("CFGdir", CFGDir, RegistryValueKind.String);
+                regkey.SetValue("MinimizeOnVMStart", MinimizeOnVMStart, RegistryValueKind.DWord);
+                regkey.SetValue("ShowConsole", ShowConsole, RegistryValueKind.DWord);
+                regkey.SetValue("MinimizeToTray", MinimizeToTray, RegistryValueKind.DWord);
+                regkey.SetValue("CloseToTray", CloseToTray, RegistryValueKind.DWord);
+                // why not just store this as an int...i might change this
+                regkey.SetValue("LaunchTimeout", Convert.ToInt32(LaunchTimeout), RegistryValueKind.DWord);
+                regkey.SetValue("EnableLogging", EnableLogging, RegistryValueKind.DWord);
+                regkey.SetValue("LogPath", LogPath, RegistryValueKind.String);
+                regkey.SetValue("EnableGridLines", EnableGridLines, RegistryValueKind.DWord);
+                regkey.Close();
+
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("An error has occurred. Please provide the following information to the developer:\n" + ex.Message + "\n" + ex.StackTrace, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns the 86Box build number as a 
+        /// </summary>
+        /// <returns></returns>
+        public static FileVersionInfo Get86BoxVersion() => FileVersionInfo.GetVersionInfo($@"{EXEDir}\86Box.exe");
+
+        //Checks if all controls match the currently saved settings to determine if any changes were made
+        public static bool CheckForChanges()
```

Comment:
The Apply button no longer works, there's a problem either here or in the code that uses this function's result and toggles the button's enabled property.